counsel-ag-popup.el

```diff
@@ -0,0 +1,137 @@
+;;; counsel-ag-popup.el --- Interactive search with ag -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2019-2020  Eder Elorriaga
+
+;; Author: Eder Elorriaga <gexplorer8@gmail.com>
+;; Keywords: convenience, matching, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Just call the interactive function `counsel-ag-popup' and use the
+;; popup to configure the search.
+
+;;; Code:
+
+(require 'counsel)
+(require 'transient)
+
+(eval-when-compile
+  (require 'subr-x))
+
+(defun counsel-ag-popup-search-here ()
+  "Ag search in current directory."
+  (interactive)
+  (counsel-ag-popup-search default-directory))
+
+(defun counsel-ag-popup-search (directory)
+  "Ag search in DIRECTORY."
+  (interactive "DDirectory: ")
+  (let ((ag-args (counsel-ag-popup--map-args (transient-args 'counsel-ag-popup))))
+    (counsel-ag "" directory ag-args)))
+
+(defun counsel-ag-popup--map-args (transient-args)
+  "Convert TRANSIENT-ARGS to a string of args."
+  (mapconcat
+   (lambda (arg)
+     (if (listp arg)
+         (let ((args (cdr arg)))
+           (mapconcat (lambda (x) (concat "--" x)) args " "))
+       arg))
+   transient-args
+   " "))
+
+(defconst counsel-ag-popup-file-types
```

Comment:
Why a `defconst`? I think it would be reasonable to want to modify this.  I'd also recommend adding a docstring. ;)

Reply:
I choose `defconst` because I got these values executing `ag --list-file-types` so I didn't expect the user to modify it.

Reply:
I figured, but I can imagine someone wanting to e.g. narrow down the possibilities for their exact use cases. It obviously doesn't matter much, of course.